bin/_main

```diff
@@ -150,57 +159,66 @@ function post_process_run() {
-            if [ ! -z "$this_id" ]; then
-                cdm_query_cmd="${CRUCIBLE_HOME}/subprojects/core/CommonDataModel/queries/cdmq/get-result-summary.sh"
-                cdm_query_cmd+=" --run=$this_id"
-                echo "Generating benchmark summary report"
-                sleep 5 # ES may take 5 seconds to have data ready (from a setting which can vastly improve index operations)
-                $podman_run --name crucible-get-result-$(uuidgen) "${container_common_args[@]}" "${container_rs_args[@]}" $CRUCIBLE_CONTAINER_IMAGE $cdm_query_cmd | tee "${RUN_DIR}/run/result-summary.txt"
-                RET_VAL=$?
-                if [ $RET_VAL -eq 0 ]; then
-                    echo
-                    echo "Benchmark summary is complete and can be found in:"
-                    echo "${RUN_DIR}/run/result-summary.txt"
-                else
-                    echo "ERROR: Could not generate benchmark summary"
-                fi
+    if $podman_run --name crucible-rickshaw-index-$(uuidgen) "${container_common_args[@]}" "${container_rs_args[@]}" $CRUCIBLE_CONTAINER_IMAGE $rs_index_cmd; then
+        echo "Benchmark result now in elastic, localhost:9200"
+        this_id=`$podman_run --name crucible-extract-run-id-$(uuidgen) "${container_common_args[@]}" "${container_rs_args[@]}" $CRUCIBLE_CONTAINER_IMAGE /bin/bash -c "${result_dumper} ${run_json} | jq -r '. \"run-id\"'"`
+        if [ ! -z "$this_id" ]; then
+            cdm_query_cmd="${CRUCIBLE_HOME}/subprojects/core/CommonDataModel/queries/cdmq/get-result-summary.sh"
+            cdm_query_cmd+=" --run=$this_id"
+            echo "Generating benchmark summary report"
+            sleep 5 # ES may take 5 seconds to have data ready (from a setting which can vastly improve index operations)
+            $podman_run --name crucible-get-result-$(uuidgen) "${container_common_args[@]}" "${container_rs_args[@]}" $CRUCIBLE_CONTAINER_IMAGE $cdm_query_cmd | tee "${RUN_DIR}/run/result-summary.txt"
+            RC=$(echo ${PIPESTATUS[@]} | sed 's/ /+/g' | bc)
+            if [ ${RC} == 0 ]; then
+                echo
+                echo "Benchmark summary is complete and can be found in:"
+                echo "${RUN_DIR}/run/result-summary.txt"
+                return 0
             else
-                echo "ERROR: Could not get the run-id from this run"
+                echo "ERROR: Could not generate benchmark summary"
+                return 1
             fi
         else
-            echo "ERROR: Could not index result into elasticsearch"
+            echo "ERROR: Could not get the run-id from this run"
+            return 1
         fi
     else
-        echo "ERROR: Could not find result json '${run_json}'"
+        echo "ERROR: Could not index result into elasticsearch"
+        return 1
```

Comment:
return code 1 may be convenient for the time being, however it might be a problem to test and get the expected result... You may need to grep for the stdout to see if your code is behaving properly....

Reply:
ack